v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -248,15 +248,43 @@ func (r *azureDeploymentReconcilerInstance) MonitorDelete(ctx context.Context) (
 	return ctrl.Result{Requeue: true, RequeueAfter: retryAfter}, nil
 }
 
-func (r *azureDeploymentReconcilerInstance) BeginCreateOrUpdateResource(ctx context.Context) (ctrl.Result, error) {
+func (r *azureDeploymentReconcilerInstance) BeginCreateOrUpdateResource(
+	ctx context.Context,
+) (ctrl.Result, error) {
 	if r.Obj.AzureName() == "" {
 		err := errors.New("AzureName was not set. A webhook should default this to .metadata.name if it was omitted. Is the ASO webhook service running?")
-		return ctrl.Result{}, conditions.NewReadyConditionImpactingError(err, conditions.ConditionSeverityError, conditions.ReasonFailed)
+		return ctrl.Result{},
+			conditions.NewReadyConditionImpactingError(
+				err, conditions.ConditionSeverityError, conditions.ReasonFailed)
+	}
+
+	check, err := r.preReconciliationCheck(ctx)
+	if err != nil {
+		// Failed to do the pre-reconciliation check, this is a serious but non-fatal error
+		// Make sure we return a ReadyConditionImpactingError so that the Ready condition is updated for the user
+		// Ideally any implementation of the checker should return a ReadyConditionImpactingError, but we can't
+		// guarantee that, so we wrap as required
+		impactingError, ok := conditions.AsReadyConditionImpactingError(err)
+		if !ok {
+			impactingError = conditions.NewReadyConditionImpactingError(
+				err,
+				conditions.ConditionSeverityWarning,
+				conditions.ReasonFailed)
+		}
+
+		return ctrl.Result{}, impactingError
+	}
+
+	// If the check says we don't need to reconcile, we're done
```

Comment:
minor: Clarify this, as I think you mean "we're done for now?" -- since if you were done we wouldn't be returning a `ConditionImpactingError`

Reply:
Ah, see my comment below in `prereconciliation_checker` - I think there's an issue w/ how you're handling the `Postpone` case here. `CreateConditionError` should return `nil` in the `Postpone` case.

Reply:
I've modified the check result to properly do this.